tests/trainer/test_checkpoint.py

```diff
@@ -141,10 +142,19 @@ def _assert_weights_equivalent(self, m1: torch.nn.Module, m2: torch.nn.Module):
         for p1, p2 in zip(m1.parameters(), m2.parameters()):
             torch.testing.assert_close(p1, p2)
 
-    def get_trainer(self, **kwargs):
+    def get_trainer(self, checkpoint_save_path=None, overwrite_checkpoints=None, **kwargs):
```

Comment:
Explicit over implicit: instead, just have the tests calling this to create the checkpointsaver themselves, and pass it in as `callbacks` to `get_trainer`.

Then `get_trainer` just does something like:
```
if 'calbacks' in kwargs:
    callbacks = kwargs.pop('callbacks')
    callbacks.append([DummyStatefulCallback()])
else:
    callbacks = [DummyStatefulCallback()]
```
    
```

Reply:
Oh nice. Good idea! Dependency injection. Done.